### PR TITLE
Frontend admin setup fixes

### LINKED ISSUE #2851: Don't show "recommended next step" if there is no recommended next step
screenshot 2016-06-29 17 47 49

### LINKED ISSUE #2906: Getting set up steps don't route
If this is a bug report, please fill in the blanks:

*  I am using the Chrome browser.
*  My computer's OS is Mac OS X Yosemite.
*  I'm running Metabase via `lein ring server`. [locally from the JAR / locally with the Mac App / on AWS / on Herkou / etc.]
*  My database is Postgres.
*  My Metabase version is 19.0-snapshot.

Getting set up "Next steps"  `email` and `slack` don't seem to route correctly until I click another step and then hit "back"

screen shot 2016-07-06 at 6 27 49 pm

### PR DESCRIPTION
**Pull request by tlrobinson**
- Resolves #2851 
- Resolves #2906 
- cleanup some Redux stuff.

### CODE REVIEW

File: src/metabase/api/setup.clj [hunk 1]
```diff
@@ -92,7 +92,7 @@
 ;;; Admin Checklist
 
 (defn- admin-checklist-values []
-  (let [has-dbs?           (db/exists? Database)
+  (let [has-dbs?           (db/exists? Database, :is_sample false)
```
Comment (camsaul) [inline src/metabase/api/setup.clj]: I think I already did this in #2901

Comment (tlrobinson) [inline src/metabase/api/setup.clj]: Yeah I accidentally branched off that. Should merge fine though.

**Comment by tlrobinson** [comment]
@kdoh can I consider your closing of #2895 in favor of this one a 👍 or should I have someone else review it?

**Comment by kdoh** [comment]
LGTM

**Pull request merged**